### PR TITLE
Add influxdb-metrics integration

### LINKED ISSUE #85: Add new influxdb/influxdb-metrics integration(s)
TODO: 

- [ ] Init integration from template (see: https://github.com/sensu/catalog/blob/main/templates)
  - [ ] Add `README.md`
  - [ ] Add `sensu-integration.yaml`
  - [ ] Add `CHANGELOG.md`
- [ ] Add `logo.png`
- [ ] Add `sensu-resources.yaml`
  - [ ] [influxdb/influxdb-legacy](https://github.com/sensu/catalog/tree/ch/flatten-the-catalog/integrations/influxdb/influxdb-legacy.yaml) 
  - [ ] [influxdb/influxdb](https://github.com/sensu/catalog/tree/ch/flatten-the-catalog/integrations/influxdb/influxdb.yaml) 
  - [ ] add a pipeline resource (example: [pagerduty/pagerduty-incidents](https://github.com/sensu/catalog/blob/main/integrations/pagerduty/pagerduty-incidents/sensu-resources.yaml#L1-L21))
  - [ ] move configured filters and mutators (if any) from handler(s) to the pipeline resource

### PR DESCRIPTION
**Pull request by ccressent**
This is still a WIP, especially when it comes to environment variables vs
secrets and content of the README.

Closes #85.

### CODE REVIEW

File: integrations/influxdb/influxdb-metrics/sensu-integration.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      input:
+        type: string
+        required: true
+        title: InfluxDB User
+        description: >-
+          Some useful description of what to put in here
+        default: ""
+
+    - type: question
+      name: influxdb_password_secret_provider
```
Comment (c-kruse) [inline integrations/influxdb/influxdb-metrics/sensu-integration.yaml:56]: Name does not match reference on line 80

**Review by c-kruse — COMMENTED** [review]
Having a rough time trying to operate influxdb myself. Is this meant to work with the relatively new influxdb 2.0 or is it 1.x only?

**Comment by ccressent** [comment]
> Having a rough time trying to operate influxdb myself. Is this meant to work with the relatively new influxdb 2.0 or is it 1.x only?

The [`sensu-influxdb-handler` plugin](https://github.com/sensu/sensu-influxdb-handler) uses [influxdata/influxdb1-client](https://github.com/influxdata/influxdb1-client), so it's for InfluxDB 1.7 and earlier; for 1.8 and 2.x, one is supposed to use [influxdata/influxdb-client-go](https://github.com/influxdata/influxdb-client-go).

There is already an issue open about 1.8/2.x support: https://github.com/sensu/sensu-influxdb-handler/issues/38. We might want to address it if we want this catalog item to be the best it can be at release...

**Comment by calebhailey** [comment]
Rebased against main

**Comment by calebhailey** [comment]
Related: https://github.com/sensu/sensu-enterprise-go/pull/2182

**Comment by calebhailey** [comment]
Rebased against main

**Comment by calebhailey** [comment]
...and rebased again. Going to get this reviewed and merged now.

**Pull request merged**